index.html

```diff
@@ -2175,6 +2231,130 @@ <h3>Verify Proof Sets and Chains</h3>
         </ol>
       </section>
 
+      <section class="normative">
+        <h3>Context Validation</h3>
+
+        <p>
+The purpose of the algorithm in this section is to ensure that a consuming
+application has explicitly approved of the types, and therefore the semantics,
+of input documents that it will process. Not checking JSON-LD context values
+against known good values can lead to security vulnerabilities, due to variance
+in the semantics that they convey. The context validation algorithm is one
+that applications validating a [=conforming secured document=] MUST run after
+running the algorithm in Section [[[#verify-proof]]] or Section
+[[[#verify-proof-sets-and-chains]]]. This algorithm takes a document ([=map=]
+|inputDocument|), a set of approved JSON-LD Contexts ([=map=]
+|approvedContexts|), and a boolean to recompact when unknown contexts are
+detected ([=boolean=] |recompact|) and returns a [=map=] that contains the
+following:
+        </p>
+
+        <ul>
+          <li>
+a status ([=boolean=] |status|)
+          </li>
+          <li>
+a validated document ([=map=] |document|) if no errors occurred
+          </li>
+          <li>
+zero or more warnings ([=list=] of [=ProblemDetails=] |warnings|)
+          </li>
+          <li>
+zero or more errors ([=list=] of [=ProblemDetails=] |errors|)
+          </li>
+        </ul>
+
+        <p>
+The context validation algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set |result|.|status| to `false`, |result|.|warnings| to an empty list,
+|result|.|errors| to an empty list, |compactionContext| to an empty list,
+and clone |inputDocument| to |result|.|document|.
```

Comment:
```suggestion
Set |result|.|status| to `false`, |result|.|errors| to an empty list,
|compactionContext| to an empty list, and clone |inputDocument|
to |result|.|document|.
```

Reply:
Including suggestion from @dlongley... This, or the following...
```suggestion
Set |result|.|status| to `false`, |result|.|warnings| to an empty list,
|result|.|errors| to an empty list, |compactionContext| to an empty list,
and |inputDocument| to a clone of |result|.|document|.
```

Reply:
Including suggestion from @dlongley... This, or the above...
```suggestion
Set |result|.|status| to `false`, |result|.|warnings| to an empty list,
|result|.|errors| to an empty list, |compactionContext| to an empty list;
and clone |inputDocument| to |result|.|document|.
```